src/seedu/addressbook/data/person/Address.java

```diff
@@ -36,6 +38,22 @@ public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    /**
+     * Void function that splits the address into block, street, postal code, email
+     */
+    
+    private void splitAddress(String addressInput) throws IllegalValueException {
+        String differentAddressParts[] = addressInput.split(", ");
+        
+        if(differentAddressParts.length != 4)
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        
+        Block block = new Block(differentAddressParts[0]);
+        Street street = new Street(differentAddressParts[1]);
+        Unit email = new Unit(differentAddressParts[2]);
+        PostalCode postalCode = new PostalCode(differentAddressParts[3]);
```

Comment:
Can remove the magic numbers. Based on the UML diagram, the new variables should be declared as class attributes.